src/typeguard/_transformer.py

```diff
@@ -500,17 +505,27 @@ def __init__(
         self.target_lineno = target_lineno
 
     def generic_visit(self, node: AST) -> AST:
-        non_empty_list_fields = []
-        for field_name, val in iter_fields(node):
-            if isinstance(val, list) and len(val) > 0:
-                non_empty_list_fields.append(field_name)
+        class HasBody(Protocol):
+            body: list[stmt]
+
+        def has_body(node: AST) -> TypeGuard[HasBody]:
+            return hasattr(node, "body") and node.body is not None
```

Comment:
Why all this here? What value does it add?

Reply:
I just avoid adding these checks in my if statements. I can remove it if you prefer.

Reply:
But your if statements would run these checks anyway? I just don't understand why this is somehow better than `if getattr(node, "body", None):`.

Reply:
The `if` statement would run these checks anyway but I only have to write on check now. I will rewrite without.

Reply:
Changed